client/ayon_core/pipeline/create/context.py

```diff
@@ -1468,6 +1467,267 @@ def execute_autocreators(self):
         if failed_info:
             raise CreatorsCreateFailed(failed_info)
 
+    def get_folder_entities(self, folder_paths: Iterable[str]):
+        """Get folder entities by paths.
+
+        Args:
+            folder_paths (Iterable[str]): Folder paths.
+
+        Returns:
+            Dict[str, Optional[Dict[str, Any]]]: Folder entities by path.
+
+        """
+        output = {
+            folder_path: None
+            for folder_path in folder_paths
+            if folder_path is not None
+        }
+        remainder_paths = set()
+        for folder_path in output:
+            # Skip empty/invalid folder paths
+            if folder_path is None or "/" not in folder_path:
+                continue
+
+            if folder_path not in self._folder_id_by_folder_path:
+                remainder_paths.add(folder_path)
+                continue
+
+            folder_id = self._folder_id_by_folder_path.get(folder_path)
+            if not folder_id:
+                output[folder_path] = None
+                continue
+
+            folder_entity = self._folder_entities_by_id.get(folder_id)
+            if folder_entity:
+                output[folder_path] = folder_entity
+            else:
+                remainder_paths.add(folder_path)
+
+        if not remainder_paths:
+            return output
```

Comment:
Looking at all of the other logic and queries done here - why do we need the folder_entity_by_id? Why can't just store `folder_entity_by_folder_path`?

That would basically make this logic:
```
output = {}
not_cached = set()
for folder_path in folder_paths:
    if folder_path in self._folder_entity_by_folder_path:
        output[folder_path] = self._folder_entity_by_folder_path[folder_path]
    else:
        not_cached.add(folder_path)
```
Right?

Reply:
Very good point, simplified.